.github/workflows/build-push-images-pr.yaml

```diff
@@ -4,15 +4,113 @@ on:
     branches:
       - develop
 jobs:
+  lint:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version:
+          - 3.7
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+
+      - name: Test installing development dependencies
+        working-directory: deployment/docker
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r REQUIREMENTS-dev.txt
+
+      - name: Run Flake8 test
+        run: flake8 .
+
+  test:
+    runs-on: ubuntu-latest
+    needs:
+      - lint
+    env:
+      APP_IMAGE: kartoza/projecta-uwsgi
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Build test image
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: deployment/docker/Dockerfile
+          push: false
+          load: true
+          target: dev
+          tags: ${{ env.APP_IMAGE }}:dev
+          cache-from: |
+            type=gha,scope=test
+            type=gha,scope=prod
+          cache-to: type=gha,scope=test
+
+      - name: Build prod image
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: deployment/docker/Dockerfile
+          push: false
+          load: true
+          target: dev
+          tags: ${{ env.APP_IMAGE }}:prod
+          cache-from: |
+            type=gha,scope=test
+            type=gha,scope=prod
+          cache-to: type=gha,scope=prod
+
+      - name: Run docker-compose services
+        working-directory: deployment
+        run: |
+          echo "Override docker-compose for testing purposes"
+          cp .env.example .env
+          cp docker-compose.test.yml docker-compose.override.yml
+          make up
+          make devweb
+          make wait-db
+          make create-test-db
+
+      - name: Run Coverage test
+        working-directory: deployment
+        run: |
+          cat << EOF | docker-compose exec -T devweb bash
+          python manage.py makemigrations
+          python manage.py migrate
+          python manage.py collectstatic --noinput --verbosity 0
+          coverage run manage.py test
+          coverage xml
+          EOF
+          docker cp projecta_devweb_1:/home/web/django_project/coverage.xml ../coverage.xml
+
+      - name: Upload coverage to codecov
+        env:
+          CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
+        if: env.CODECOV_TOKEN != ''
```

Comment:
Contributor fork might not have CODECOV integration, so we skip this step in that case.